akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/RetentionCriteriaImpl.scala

```diff
@@ -26,14 +26,14 @@ import akka.persistence.typed.scaladsl
 
   def deleteUpperSequenceNr(lastSequenceNr: Long): Long = {
     // Delete old events, retain the latest
-    math.max(0, lastSequenceNr - (keepNSnapshots * snapshotEveryNEvents))
+    math.max(0, lastSequenceNr - (keepNSnapshots.asInstanceOf[Number].longValue() * snapshotEveryNEvents))
```

Comment:
```suggestion
    math.max(0, lastSequenceNr - (keepNSnapshots.toLong * snapshotEveryNEvents))
```
And for the other one as well

Reply:
Thanks!